### ISSUE TITLE
Add optional consent due to GDPR

### ISSUE THREAD

**Issue by lknik** [w3c/permissions#169]
We had a discussion during London F2F  (https://github.com/w3ctag/meetings/tree/gh-pages/2018/01-london) about adding optional purpose definition during permission requests.

The idea is that a developers could annotate requests for e.g. access to geolocation with some comment such as purpose. It would now reflect a bit how iOS permission requests work. Could simplify consents/etc.

Input welcome.

**Comment by dbaron** [comment]
The [minutes of the discussion](https://cryptpad.w3ctag.org/code/#/1/view/QSL8iYiATXlRsxx7ITX5Nw/gVwKfF8uYFYFjF-k6U73QCx9odKA1ohLMtrHerICR2I/) (see the section "EU GDPR/ePrivacy Regulation and browsers") may be useful for understanding.

**Comment by raymeskhoury** [comment]
This is something that we've considered, but I don't think we would ever expose the site-provided request reason in Chrome UI. As was noted in the meeting notes, placing site-provided data in trusted UI has the potential to mislead users into believing that the reason given is trustworthy, when in fact it may not be. The same argument applies even with a predefined set of reasons why the data may be needed. The site can say "I need your location so I can show you your position on the map" but there's nothing stopping it from also using it for ad targeting.

**Comment by johannhof** [comment]
I think from Firefox side we will have objections to putting arbitrary text in browser chrome UI, too, for the same reasons (it looks like dbaron mentioned that in the meeting already).

Many of the bigger websites have adopted a pre-prompt pattern (some element on the site that indicates that the site would like to ask for permissions) and evangelizing that seems like a better option to me.

**Comment by tcellaaron** [comment]
I agree with @johannhof as long as a page can always check whether an action it will cause a permission request they can build ui to make clear what they are trying to do before the user ever sees a permission prompt.
seems like the important thing is that sites can know what permissions they have/would need to ask for.
Are developers having problems implementing this pattern?
